File: javascript-leetcode/746.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
  let f0 = cost[0];
  let f1 = cost[1];
  let f2 = cost[2] + Math.min(f0, f1);
  for (let i = 3; i < cost.length; i += 1) {
    [f0, f1, f2] = [f1, f2, cost[i] + Math.min(f1, f2)];
  }
  return Math.min(f1, f2);
};

File: python-leetcode/746.py
class Solution:
  def minCostClimbingStairs(self, cost: List[int]) -> int:
    # dp[i] = cost[i] + min(dp[i-1], dp[i-2])
    # result = min(dp[-1], dp[-2])
    f1, f2 = cost[0], cost[1]
    for i in range(2, len(cost)):
      f1, f2 = f2, cost[i] + min(f1, f2)
    return min(f1, f2)

